sdk/include/stdlib.h

```diff
@@ -291,17 +291,36 @@ ssize_t __cheri_compartment("allocator")
   heap_quota_remaining(AllocatorCapability heapCapability);
 
 /**
- * Block until the quarantine is empty.
+ * Try to empty the quarantine and defragment the heap.
  *
- * This should be used only in testing, to place the system in a quiesced
- * state.  It can block indefinitely if another thread is allocating and
- * freeing memory while this runs.
+ * This will start a revocation sweep and then try to empty the quarantine.  In
+ * normal operation, the allocator will remove a small number of allocations
+ * from quarantine on each allocation.  Allocations are not coalesced until
+ * they are moved from quarantine, so this can cause fragmentation.  If you
+ * have just freed a lot of memory (for example, after resetting a compartment
+ * and calling `heap_free_all`), especially if you have freed a lot of small
+ * allocations, then calling this function will reduce fragmentation.
```

Comment:
I don't think it's a certainty?

```suggestion
 * allocations, then calling this function will likely reduce fragmentation.
```